rules/rules.js

```diff
@@ -85,21 +116,27 @@ const getGroupsForItem = function (ruleConfig) {
  *
  * rules.JSRule({
  *  name: "my_new_rule",
- *  description": "this rule swizzles the swallows",
+ *  description: "this rule swizzles the swallows",
  *  triggers: triggers.GenericCronTrigger("0 30 16 * * ? *"),
  *  execute: triggerConfig => { //do stuff }
  * });
  *
  * @memberOf rules
  * @param {Object} ruleConfig The rule config describing the rule
  * @param {String} ruleConfig.name the name of the rule
- * @param {String} ruleConfig.description a description of the rule
+ * @param {String} [ruleConfig.description] a description of the rule
  * @param {*} ruleConfig.execute callback that will be called when the rule fires
  * @param {HostTrigger|HostTrigger[]} ruleConfig.triggers triggers which will define when to fire the rule
+ * @param {String} [ruleConfig.id] the UID of the rule
+ * @param {String} [ruleConfig.ruleGroup] the name of the rule group to use
+ * @param {Boolean} [ruleConfig.overwrite=false] overwrite an existing rule with the same UID
  * @returns {HostRule} the created rule
  */
 let JSRule = function (ruleConfig) {
     let ruid = ruleConfig.id || ruleConfig.name.replace(/[^\w]/g, "-") + "-" + utils.randomUUID();
+    if (ruleConfig.overwrite === true) {
```

Comment:
So without this option, if a rule exists with the same ID , it would thrown an error when trying to persist it?

Reply:
No, it would just be quiet without that option set.

Reply:
huh, thats not actually what i thought would happen :-) is that desired?  i'm thinking we should always then remove the rule if it exists, or thrown an error. So right now, you would have no idea if it save properly or not without this flag right?

Reply:
I don't think is desired that there is no error, but I am watching this behavious on my system.

I will add something to throw an error as overriding by default is nothing @jpg0 would do.